### ISSUE TITLE
Dialogs leave black rectangles

### ISSUE THREAD

**Issue by drguildo** [processing/processing#4719]
Sometimes after closing a dialog or menu that's drawn over the main window, a black rectangle is left in it's place. Minimising and restoring the window (to force a complete redraw of the window) gets rid of the rectangle.

![image](https://cloud.githubusercontent.com/assets/9282/19771109/abfa635e-9c59-11e6-8f41-1ec93afb60c1.png)

**Comment by benfry** [comment]
I've not seen this… Likely something weird with your graphics drivers, but we'll keep an eye out.

**Comment by drguildo** [comment]
Yeah, I thought as much and wouldn't have reported it except I've never experienced the same issue with any other programs.

If it's of any use I have an NVIDIA GeForce GTX 660 with version 368.81 of the graphics driver.

**Comment by oth-aw** [comment]
We have the same problems with our windows machines.

**Comment by benfry** [comment]
What version of Windows? 32- or 64-bit? Latest version of Processing (3.2.2)? Anything else you can tell me about the machines?

**Comment by oth-aw** [comment]
I am not at work now. But I guess: Processing 3.2.1 Windows 10 with 64 bit.

**Comment by drguildo** [comment]
> What version of Windows? 32- or 64-bit? Latest version of Processing (3.2.2)? Anything else you can tell me about the machines?

Windows 7 64-bit here. Processing 3.2.2 but the problem has existed for as long as I can remember across multiple Processing and GPU driver versions.

**Comment by benfry** [comment]
Meaning since 3.x? Since 2.x?

**Comment by drguildo** [comment]
> Does "as long as you remember" mean since 3.x? Since 2.x?

I think since 2.x. Sorry about being vague.

> On closer look, it seems that it's limited to the text area—is that correct? i.e. there's no black on the scrollbars and the rest of the panels in the window are all refreshed properly?

It seems to be, yeah.
